lib/IRGen/TBDGen.cpp

```diff
@@ -472,9 +472,15 @@ void TBDGenVisitor::addProtocolWitnessThunk(RootProtocolConformance *C,
                                             ValueDecl *requirementDecl) {
   Mangle::ASTMangler Mangler;
 
+  std::string decorated = Mangler.mangleWitnessThunk(C, requirementDecl);
   // FIXME: We should have a SILDeclRef SymbolSource for this.
-  addSymbol(Mangler.mangleWitnessThunk(C, requirementDecl),
-            SymbolSource::forUnknown());
+  addSymbol(decorated, SymbolSource::forUnknown());
+
+  ValueDecl *PWT = C->getWitness(requirementDecl).getDecl();
+  if (fixmeWitnessHasLinkageThatNeedsToBePublic(SILDeclRef(PWT)))
```

Comment:
I think this check is likely redundant with the check in `lib/SIL/IR/SILSymbolVisitor.cpp` that guards calling `addProtocolWitnessThunk`.

If the function (`decorated` above) is public the async function pointer is also public.